### PR TITLE
keycloak: Add option to create authentication sub-flow of type 'form flow'

### PR DESCRIPTION
**Pull request by flyingflo**
##### SUMMARY

To create something like keycloak's built-in registration flow,
we need to create a subflow with the type 'form-flow'.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

keycloak_authentication
##### ADDITIONAL INFORMATION
Additionally, this adds some detail to the error message created on keycloak API exceptions, i.e.
fields from the HTTPError exception. This is optional.

Another patch fixes an Exception when formatting an error message.

```paste below

```

### CODE REVIEW

File: changelogs/fragments/6318-add-form-flow.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+bugfixes:
+  - "keycloak - Improve error messages (https://github.com/ansible-collections/community.general/pull/6318)."
```
Comment (felixfontein) [inline changelogs/fragments/6318-add-form-flow.yml:2]:
```suggestion
  - "keycloak - improve error messages (https://github.com/ansible-collections/community.general/pull/6318)."
```

File: changelogs/fragments/6318-add-form-flow.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+minor_changes:
+  - "keycloak_authentication - Add flow type option to sub flows to allow the creation of 'form-flow' sub flows like in keycloak's built-in registration flow (https://github.com/ansible-collections/community.general/pull/6318)."
```
Comment (felixfontein) [inline changelogs/fragments/6318-add-form-flow.yml:5]:
```suggestion
  - "keycloak_authentication - add flow type option to sub flows to allow the creation of 'form-flow' sub flows like in Keycloak's built-in registration flow (https://github.com/ansible-collections/community.general/pull/6318)."
```

File: plugins/modules/keycloak_authentication.py [hunk 3]
```diff
@@ -79,6 +79,12 @@
                 description:
                     - Priority order of the execution.
                 type: int
+            subFlowType:
+                description:
+                    - For new subflows, optionally specify the type.
```
Comment (felixfontein) [inline plugins/modules/keycloak_authentication.py:84]: If this is only used at creation, you should mention that, like:
```suggestion
                    - For new subflows, optionally specify the type.
                    - Is only used at creation.
```

File: plugins/modules/keycloak_authentication.py [hunk 4]
```diff
@@ -79,6 +79,12 @@
...
+                choices: ["basic-flow", "form-flow"]
+                default: "basic-flow"
+                type: str
```
Comment (felixfontein) [inline plugins/modules/keycloak_authentication.py:87]:
```suggestion
                type: str
                version_added: 6.6.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

**Comment by felixfontein** [comment]
Please also check the failing unit tests.

**Comment by flyingflo** [comment]
> Please also check the failing unit tests.

The fail was a good catch. We must exclude the new parameter from checks, because it is only valid for creation.

**Comment by flyingflo** [comment]
Sorry for messing up the PR history.. I forgot to pull the changes commited in the web ui.

File: changelogs/fragments/6318-add-form-flow.yml (hunk 2 above)
Comment (felixfontein) [reply changelogs/fragments/6318-add-form-flow.yml:5]: This one is still open.

**Review by felixfontein — APPROVED** [review]
Looks good to me as far as I can judge. I'll merge this this weekend if nobody objects.

**Comment by felixfontein** [comment]
@flyingflo thanks for your contribution!

**Pull request merged**